docs/FIELDS.md

```diff
@@ -0,0 +1,338 @@
+  - application.properties
+  - more.properties
+```
+
+### `secretGenerator`
+
+Each entry in this list results in the creation of
+one Secret resource (it's a generator of n secrets).
+A command can do anything to get a secret,
+e.g. prompt the user directly, start a webserver to
+initate an oauth dance, etc.
+
+Usage:
+
+```yaml
+secretGenerator:
+- name: app-tls
+  commands:
+    tls.crt: "cat secret/tls.cert"
+    tls.key: "cat secret/tls.key"
+  type: "kubernetes.io/tls"
+- name: downloaded_secret
+  # timeoutSeconds specifies the number of seconds to
+  # wait for the commands below. It defaults to 5 seconds.
+  timeoutSeconds: 30
+  commands:
+    username: "curl -s https://path/to/secrets/username.yaml"
+    password: "curl -s https://path/to/secrets/password.yaml"
+  type: Opaque
+- name: env_file_secret
+  # envCommand is similar to command but outputs lines of key=val pairs
+  # i.e. a Docker .env file or a .ini file.
+  # you can only specify one envCommand per secret.
+  envCommand: printf "DB_USERNAME=admin\nDB_PASSWORD=somepw"
+  type: Opaque
+```
+
+## Input resources
+
+### `resources`
+
+Each entry in this list must resolve to an existing
+resource definition in YAML.  These are the resource
+files that kustomize reads, modifies and emits as a
+YAML string, with resources separated by document
+markers ("---").
+
+
+Usage:
+
+```yaml
+resources:
+- some-service.yaml
+- ../some-dir/some-deployment.yaml
+```
+
+### `bases`
+
+Each entry in this list should resolve to a directory
+containing a kustomization file, else the
+customization fails.
+
+The entry could be a relative path pointing to a local directory
+or a url pointing to a directory in a remote repo.
+The url should follow hashicorp/go-getter URL format
+https://github.com/hashicorp/go-getter#url-format
+
+The presence of this field means this file (the file
+you a reading) is an _overlay_ that further
+customizes information coming from these _bases_.
+
+Typical use case: a dev, staging and production
+environment that are mostly identical but differing
+crucial ways (image tags, a few server arguments,
+etc. that differ from the common base).
+
+Usage:
+
+```yaml
+bases:
+- ../../base
+- github.com/kubernetes-sigs/kustomize//examples/multibases?ref=v1.0.6
+- github.com/Liujingfang1/mysql
+- github.com/Liujingfang1/kustomize//examples/helloWorld?ref=test-branch
+```
+
+### `crds`
+
+Each entry in this list should be a relative path to
+a file for custom resource definition(CRD).
+
+The presence of this field is to allow kustomize be
+aware of CRDs and apply proper
+transformation for any objects in those types.
+
+Typical use case: A CRD object refers to a ConfigMap object.
+In kustomization, the ConfigMap object name may change by adding namePrefix or hashing
+The name reference for this ConfigMap object in CRD object need to be
+updated with namePrefix or hashing in the same way.
+
+Usage:
+
+```yaml
+crds:
+- crds/typeA.yaml
+- crds/typeB.yaml
+```
+
+## Patching
```

Comment:
`Patching` is also confusing. I suggest to remove this category.

Reply:
or use _operators_